design/open/$PULLS.md

```diff
@@ -0,0 +1,108 @@
+# $PULLS Proposal
+
+This proposal tries to bridge between two proposals: [pull-request](pull-request.md) and [checks](https://github.com/treeverse/lakeFS/blob/280c5b741df5e1e051958b8adde826364c6df614/design/open/branch_checks.md)
+and provide long-running CI/CD processes while laying the foundation for future "pull-request" like feature.
+
+## How it will work
+
+Introduce a new entity called $PULLS at the repository level. The $PULLS entity is identified by a source branch and destination branch. 
+Each source-destination pair can have only a single `OPEN` $PULLS instance.
+Define a new hook type - `on-pull` which will run whenever a new $PULLS instance is created.
+The actions will run on the source branch, and will allow merging after job is successful.
+
+### $PULLS Object
+
+$PULLS will be saved in the ref-store under the key: 
+`$PULLS/<src-branch>/<dest-branch>/<$PULLS-id>`
+
+The $PULLS object will contain the following data:
+
+* `Source Branch`
+* `Source Ref` - The source branch head at the time of creation
+* `Destination Branch`
+* `Run ID` - The actions associated run id, may be empty
+
+### $PULLS States
+
+$PULLS will have to following states:
+
+* `OPEN`
+* `CLOSED`
+* `MERGED`
+
+### Creating a new $PULLS instance
+
+POST `/api/v1/repositories/{repository}/branches/{sourceBranch}/$PULLS/{destinationBranch}`
+
+On creation the following occurs:
+
+1. Check for conflicts between source and destination branch - if conflict is found, do not run actions and display an appropriate message
+2. Run actions - and associate RunID with instance
```

Comment:
are these the same actions we run when wanting to merge pull?

Reply:
No - merge actions run on the `pre-merge` and `post-merge` hooks .
These actions run on `on-pull` hook which is triggered when creating a new $PULLS